src/libs/actions/Task.ts

```diff
@@ -652,11 +652,16 @@ function setAssigneeChatReport(chatReport: OnyxTypes.Report) {
  * If there is no existing chat, it creates an optimistic chat report
  * It also sets the shareDestination as that chat report if a share destination isn't already set
  */
-function setAssigneeValue(assigneeEmail: string, assigneeAccountID: number, shareDestination: string, isCurrentUser = false): OnyxEntry<OnyxTypes.Report> {
+function setAssigneeValue(assigneeEmail: string, assigneeAccountID: number, shareToReportID: string, isCurrentUser = false, report: OnyxEntry<OnyxTypes.Report>): OnyxEntry<OnyxTypes.Report> {
     let chatReport: OnyxEntry<OnyxTypes.Report> = null;
 
     if (!isCurrentUser) {
-        chatReport = ReportUtils.getChatByParticipants([assigneeAccountID]);
+        if(report) {
+            chatReport = report;
+        }
+        else {
+            chatReport = ReportUtils.getChatByParticipants([assigneeAccountID]);
```

Comment:
Not sure how I feel about this. Is this correct in any real cases or is this always going to be wrong?